sdk/java/pulumi/src/test/java/com/pulumi/resources/ResourceTest.java

```diff
@@ -54,35 +62,61 @@ void testProviderPropagation() {
         assertThat(provider.get().getResourceType()).isEqualTo("pulumi:providers:test");
     }
 
-    public static class MyStack {
-        public static void init(Context ctx) {
-            var mod = "test";
-            var provider = new ProviderResource(
-                    mod, "testProvider", ResourceArgs.Empty, CustomResourceOptions.Empty
-            );
-
-            var resource = new CustomResource(
-                    mod + ":a/b:c", "testResource", ResourceArgs.Empty,
-                    CustomResourceOptions.builder()
-                            .provider(provider)
-                            .build()
-            );
-        }
+    @Test
+    void testReplaceOnChanges() {
+        var test = PulumiTestInternal.builder()
+                .options(TestOptions.builder().preview(false).build())
+                .monitorDecorator(Mockito::spy)
+                .mocks(new ReplaceOnChangesMocks())
+                .build();
+
+        var result = test.runTest(ctx ->
+                new MyResource("testResource", CustomResourceOptions.builder().replaceOnChanges("foo").build())
+        );
+
+        assertThat(result.resources())
+                .hasSize(2)
+                .hasExactlyElementsOfTypes(Stack.class, MyResource.class);
+
+        var resourceCaptor = ArgumentCaptor.forClass(Resource.class);
+        var requestCaptor = ArgumentCaptor.forClass(RegisterResourceRequest.class);
+        verify(test.monitor(), times(2)).registerResourceAsync(resourceCaptor.capture(), requestCaptor.capture());
+
+        assertThat(resourceCaptor.getAllValues().get(1)).isInstanceOf(MyResource.class);
+        assertThat(requestCaptor.getAllValues().get(1).getReplaceOnChangesList())
```

Comment:
OK interesting. I'm a bit less familiar with the mockist testing but I think this makes sense - this is saying that `monitor` receives a `registerResourceAsync` call with `RegisterResourceRequest` containing the "foo". So essentially it tests that "foo" propagates from `CustomResourceOptions` to `RegisterResourceRequest`. 

This seem reasonable since I guess SDK does not do anything else with the option is just propagates it - we do not need to test that the engine respects it. 

Thank you!